### ISSUE TITLE
API docs: document most used events, classes and variables contained in various NVDA objects and API handlers

### ISSUE THREAD

**Issue by josephsl** [nvaccess/nvda#6662]
Hi,

A branch of #6654:

When looking at NVDAObjects package source code 9part of NVDA Core), it is surprising to find that certain events, classes and variables used in NVDAObjects and friends does not have suitable docs. Although people who are add-on veterans would know how event_nameChange and others work, for new developers and users, their next best bet is to sort through app modules, global plugins and so on in order to learn what the events do and why classes are defined in certain modules.

Thus, as part of a larger work on API docs, I propose adding docstrings to things such as NVDAObjects.NVDAObject.event_* functions and so on, at least those that are encountered frequently in add-ons world and in other places.

Thanks.

**Comment by feerrenrut** [comment]
I'm going to give this a priority 4. We are happy to take a pull request for this, however it is not something that we are likely to work on internally (without first working in that area).

**Comment by Adriani90** [comment]
cc: @CyrilleB79, @XLTechie, @SaschaCowley this might need an updated triage.
